File: frontend-admin/src/api/auditLog.ts
import request from './request'

export type AuditActionType =
  | 'create'
  | 'update'
  | 'delete'
  | 'batch_update_status'
  | 'batch_delete'
  | 'import'

export interface AuditLog {
  id: number
  action_type: AuditActionType
  operator_id: number
  operator_name: string
  target_ids: number[] | null
  target_name: string | null
  changes: Record<string, { old: unknown; new: unknown }> | null
  details: string | null
  created_at: string
}

export interface AuditLogListParams {
  page?: number
  page_size?: number
  action_type?: AuditActionType
  start_date?: string
  end_date?: string
}

export interface AuditLogListResponse {
  items: AuditLog[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export const auditLogApi = {
  getList: (params: AuditLogListParams): Promise<AuditLogListResponse> => {
    return request.get('/audit-logs', { params })
  },
}

File: backend/app/schemas/audit_log.py
"""Audit log schemas for request/response validation."""

from pydantic import BaseModel
from typing import Optional, List, Any, Dict
from datetime import datetime

from app.models.audit_log import AuditActionType


class AuditLogResponse(BaseModel):
    """Schema for audit log response."""
    id: int
    action_type: AuditActionType
    operator_id: int
    operator_name: str
    target_ids: Optional[List[int]] = None
    target_name: Optional[str] = None
    changes: Optional[Dict[str, Any]] = None
    details: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class AuditLogListResponse(BaseModel):
    """Schema for paginated audit log list response."""
    items: List[AuditLogResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
